type_helpers.go

```diff
@@ -117,3 +128,70 @@ func SMTPAuthValue(v SMTPAuthType) *SMTPAuthType {
 func String(v string) *string {
 	return &v
 }
+
+// Unsettable is a wrapper that can be used for marshaling attributes which use
+// significant nil values, but still require `omitempty` behavior.
+//
+// This is generally useful for PATCH requests, where attributes with zero
+// values are intentionally not marshaled into the request payload.
+//
+// Helper functions are provided to help avoid pointer wrangling.
+//
+// NOTE: This type is only for use with outbound requests. It will cause errors
+// if used to unmarshal API responses, since the jsonapi module does not yet
+// support custom unmarshaling.
+
+type Unsettable[T any] struct {
```

Comment:
my preference would probably be to only export what we want users to directly interact with. if we're using the below helper methods, then I don't think we'd need to export this struct.